### PR TITLE
fix(dyn-sampling): Backend code clean up

### PR DESCRIPTION
**Pull request by andriisoldatenko**
We are consolidating server-side-sampling and dynamic-sampling flags into only dynamic-sampling. The flag is being controlled by plan

### CODE REVIEW

File: tests/sentry/api/endpoints/test_project_details.py [hunk 1]
```diff
@@ -1682,49 +1206,39 @@ def setUp(self):
...
-            assert response.data["dynamicSampling"] is None
             assert response.data["dynamicSamplingBiases"] == DEFAULT_BIASES
 
-    def test_get_dynamic_sampling_after_migrating_to_new_plan_manually_set_biases(self):
+    def test_get_dynamic_sampling_biases_manually_set_biases(self):
         """
-        Tests the case when an organization was in EA/LA and has setup previously Dynamic Sampling rules,
-        and now they have migrated to an AM2 plan, and have manipulated the bias toggles, so they should get their
-        actual bias preferences. This also ensures that they no longer receive the deprecated dynamic sampling rules.
+        Tests the case when an organization on AM2 plan, and have manipulated the bias toggles,
```
Comment (priscilawebdev) [inline tests/sentry/api/endpoints/test_project_details.py:1227]: nit: this sentence sounds a little off... do you mean

> Tests the case when an organization on AM2 plan has manipulated one of the bias toggles

?

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Review by wmak — COMMENTED** [review]
Can you update 
`src/sentry/api/serializers/models/organization.py:289` as part of this too

```diff
diff --git a/src/sentry/api/serializers/models/organization.py b/src/sentry/api/serializers/models/organization.py
index bfa4549fb7..26b26474a1 100644
--- a/src/sentry/api/serializers/models/organization.py
+++ b/src/sentry/api/serializers/models/organization.py
@@ -286,7 +286,7 @@ class OrganizationSerializer(Serializer):  # type: ignore
             # If the current request is using a customer domain, then we activate the feature for this organization.
             feature_list.add("customer-domains")

-        if "server-side-sampling" not in feature_list and "mep-rollout-flag" in feature_list:
+        if "dynamic-sampling" not in feature_list and "mep-rollout-flag" in feature_list:
             feature_list.remove("mep-rollout-flag")

         has_auth_provider = attrs.get("auth_provider", None) is not None
```

File: tests/sentry/api/endpoints/test_relay_projectconfigs_v2.py [hunk 2]
```diff
@@ -178,7 +178,6 @@ def test_relays_dyamic_sampling(
     with Feature(
         {
             "organizations:server-side-sampling": True,
```
Comment (matejminar) [inline tests/sentry/api/endpoints/test_relay_projectconfigs_v2.py:180]: Should we also kill/rename this?

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py [hunk 3]
```diff
@@ -175,7 +175,6 @@ def test_relays_dyamic_sampling(client, call_endpoint, default_project, dyn_samp
     with Feature(
         {
             "organizations:server-side-sampling": True,
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply tests/sentry/api/endpoints/test_relay_projectconfigs.py:177]: I saw this too but I believe he is doing more cleanup in a second Pull Request

[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply tests/sentry/api/endpoints/test_relay_projectconfigs.py:177]: i fixed it just about 1 min ago :)

File: tests/sentry/api/endpoints/test_relay_projectconfigs_v2.py (hunk 2 above)
Comment (andriisoldatenko) [reply tests/sentry/api/endpoints/test_relay_projectconfigs_v2.py:180]: @matejminar same here.

**Comment by andriisoldatenko** [comment]
@priscilawebdev / @matejminar  FYI
```
rg -w "organizations:server-side-sampling"
src/sentry/conf/server.py:1187:    "organizations:server-side-sampling-allow-incompatible-platforms": False,
src/sentry/features/__init__.py:156:default_manager.add("organizations:server-side-sampling-allow-incompatible-platforms", OrganizationFeature, True)
static/app/views/settings/project/dynamicSampling/index.tsx:25:            'organizations:server-side-sampling',
```

**Comment by priscilawebdev** [comment]
I checked it locally and it seems that we now only have the flag in the frontend code and this one `"organizations:server-side-sampling-allow-incompatible-platforms"` but if I remember well you want to tackle that in a separate Pull Request... so it looks good to me

**Comment by andriisoldatenko** [comment]
> I checked it locally and it seems that we now only have the flag in the frontend code and this one `"organizations:server-side-sampling-allow-incompatible-platforms"` but if I remember well you want to tackle that in a separate Pull Request... so it looks good to me

Yes.

**Review by iambriccardo — APPROVED** [review]
LGTM

File: src/sentry/dynamic_sampling/feature_multiplexer.py [hunk 4]
```diff
@@ -8,45 +8,16 @@
...
-
     def __init__(self, project: "Project"):
-        # Feature flag that informs us that relay is handling DS rules
-        self.allow_dynamic_sampling = features.has(
-            "organizations:server-side-sampling", project.organization
-        )
-        # Feature flag that informs us that the org is on the new AM2 plan and thereby have adaptive sampling enabled
+        # Feature flag that informs us that the org is on the new AM2 plan and thereby have adaptive
+        # sampling enabled
         self.current_dynamic_sampling = features.has(
```
Comment (jjbayer) [inline src/sentry/dynamic_sampling/feature_multiplexer.py:14]: nit: Rename to `self.dynamic_sampling`?

**Review by jjbayer — APPROVED** [review]
Could you add a PR description documenting why the old feature flags are not required anymore?

File: src/sentry/dynamic_sampling/feature_multiplexer.py (hunk 4 above)
Comment (andriisoldatenko) [reply src/sentry/dynamic_sampling/feature_multiplexer.py]: done @jjbayer

**Pull request merged**